lib/matplotlib/dates.py

```diff
@@ -595,7 +595,21 @@ def drange(dstart, dend, delta):
 
 def _wrap_in_tex(text):
     # Braces ensure dashes are not spaced like binary operators.
-    return '$\\mathdefault{' + text.replace('-', '{-}') + '}$'
+    p = re.compile("[a-zA-Z]+")
```

Comment:
may be easier to write using https://docs.python.org/3/library/re.html#re.sub?

Reply:
Thanks. I just pushed a new commit with `re.sub`. Indeed it is much easier this way.

Reply:
seems better, yes.